src/app/share/views/SharedGuestSignUp/ShareGuestSignUpView.test.tsx

```diff
@@ -0,0 +1,470 @@
+    const mockMnemonic = generateMnemonic(256);
+    const keys = await keysService.getKeys(mockPassword);
+    const encryptedMockMnemonic = encryptTextWithKey(mockMnemonic, mockPassword);
+    const creationDate = new Date();
+
+    const mockUser: UserSettings = {
+      uuid: 'mock-uuid',
+      email: mockEmal,
+      privateKey: keys.ecc.privateKeyEncrypted,
+      mnemonic: encryptedMockMnemonic,
+      userId: 'mock-userId',
+      name: 'mock-name',
+      lastname: 'mock-lastname',
+      username: 'mock-username',
+      bridgeUser: 'mock-bridgeUser',
+      bucket: 'mock-bucket',
+      backupsBucket: null,
+      root_folder_id: 0,
+      rootFolderId: 'mock-rootFolderId',
+      rootFolderUuid: undefined,
+      sharedWorkspace: false,
+      credit: 0,
+      publicKey: keys.ecc.publicKey,
+      revocationKey: keys.revocationCertificate,
+      keys: {
+        ecc: {
+          publicKey: keys.ecc.publicKey,
+          privateKey: keys.ecc.privateKeyEncrypted,
+        },
+        kyber: {
+          publicKey: keys.kyber.publicKey ?? '',
+          privateKey: keys.kyber.privateKeyEncrypted ?? '',
+        },
+      },
+      appSumoDetails: null,
+      registerCompleted: false,
+      hasReferralsProgram: false,
+      createdAt: creationDate,
+      avatar: null,
+      emailVerified: false,
+    };
+
+    callCount = 0;
+
+    (useSignUp as Mock).mockImplementation(() => ({
+      doRegisterPreCreatedUser: vi.fn().mockResolvedValue({
+        xUser: mockUser,
+        xToken: mockToken,
+        mnemonic: mockMnemonic,
+      }),
+    }));
+
+    const spy = vi.spyOn(userActions, 'setUser').mockImplementation((user) => {
+      return {
+        payload: user,
+        type: 'user/setUser',
+      };
+    });
+    render(<ShareGuestSingUpView />);
+    const submitButton = screen.getByRole('button');
+    fireEvent.click(submitButton);
+    await vi.waitFor(() => {
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+
+    const decryptedPrivateKey = keysService.decryptPrivateKey(keys.ecc.privateKeyEncrypted, mockPassword);
+
+    let decryptedPrivateKyberKey = '';
+    if (keys.kyber.privateKeyEncrypted)
+      decryptedPrivateKyberKey = keysService.decryptPrivateKey(keys.kyber.privateKeyEncrypted, mockPassword);
+
+    const mockClearUser: UserSettings = {
+      uuid: 'mock-uuid',
+      email: 'mock@email.com',
+      privateKey: Buffer.from(decryptedPrivateKey).toString('base64'),
+      mnemonic: encryptedMockMnemonic,
+      userId: 'mock-userId',
+      name: 'mock-name',
+      lastname: 'mock-lastname',
+      username: 'mock-username',
+      bridgeUser: 'mock-bridgeUser',
+      bucket: 'mock-bucket',
+      backupsBucket: null,
+      root_folder_id: 0,
+      rootFolderId: 'mock-rootFolderId',
+      rootFolderUuid: undefined,
+      sharedWorkspace: false,
+      credit: 0,
+      publicKey: keys.ecc.publicKey,
```

Comment:
for next times, maybe could extract the commont values to a one const and reuse it in the others